src/memo/memo.ml

```diff
@@ -282,6 +282,44 @@ end
 
 let _ = Value_id.to_dyn
 
+(* Dependencies of a value discovered so far.
+
+   We start discovering dependencies during the [restore_from_cache] step. Some
```

Comment:
It seems too generous to describe it as "discovering", since we have the list upfront, and simply end up copying that to Deps_so_far in a roundabout way. I wish this was less silly.

Reply:
Good point, I'll change the wording to more modest "accumulating" :)